source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -430,19 +430,18 @@ void OAuth2Filter::onGetAccessTokenSuccess(const std::string& access_code,
   id_token_ = id_token;
   refresh_token_ = refresh_token;
 
-  const auto new_epoch = time_source_.systemTime() + expires_in;
-  new_expires_ = std::to_string(
-      std::chrono::duration_cast<std::chrono::seconds>(new_epoch.time_since_epoch()).count());
-
-  finishFlow();
+  finishFlow(expires_in);
 }
 
-void OAuth2Filter::finishFlow() {
+void OAuth2Filter::finishFlow(std::chrono::seconds expires_in) {
   std::string token_payload;
+  auto new_expires = time_source_.systemTime() + expires_in;
+  std::string new_expires_str = std::to_string(
```

Comment:
Both of these can be `const` I think

Reply:
Done, also moved these before the definition of token_payload.